src/Integration/Notifications/NotificationIndicatorViewModel.cs

```diff
@@ -0,0 +1,175 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Windows.Input;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Integration.WPF;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.Notifications
+{
+    public class NotificationIndicatorViewModel : ViewModelBase, INotificationIndicatorViewModel
+    {
+        private string text;
+        private bool hasUnreadEvents;
+        private bool isVisible;
+        private bool isBalloonTooltipVisible;
+        private bool areNotificationsEnabled;
+        private readonly ITimer autocloseTimer;
+        private readonly Action<Action> uiThreadInvoker;
+
+        public ObservableCollection<SonarQubeNotification> NotificationEvents { get; }
+
+        // For testing
+        public NotificationIndicatorViewModel()
+            : this (ThreadHelper.Generic.Invoke)
+        {
+        }
+
+        public NotificationIndicatorViewModel(Action<Action> uiThreadInvoker)
+        {
+            this.uiThreadInvoker = uiThreadInvoker;
+
+            NotificationEvents = new ObservableCollection<SonarQubeNotification>();
+            text = GetTooltipText();
+
+            autocloseTimer = new TimerWrapper { Interval = 3000, AutoReset = false };
```

Comment:
What's the point of using the wrapper if you instantiate it directly instead of taking the interface as ctor argument? I would either use the timer itself or use the interface.

Reply:
Haha, I missed that one!